### DISCUSSION TITLE
MAC dilation with PFX

### DISCUSSION THREAD

**Discussion by Prody33** [openssl/openssl#22799] [Q&A] [answered]
Good morning,

I come to share with you my experiences with the hope of finding someone who encountered the same problem as me and who succeeded where I failed in resolving this problem.

Here I put my openssl on Windows (I know it's bad) in without new versions in 3.1.4. Previously I was running a lower version of Linux.

But when I started to create my PFX in this new version, I ran into a problem which told me when I tried to import it onto a certain Windows system that the password was not the correct one even though it was very correct.

Obviously in some Windows it worked but not others. So I discovered that it depends on the version of the OS and that the recent ones take the certificate but not the old ones.

After some research I discovered that this was due to the default encryption method used by OpenSSL which uses AES while the older ones use 3DES.

Windons in slightly older versions such as W2K12R2 do not support it.

So I overcome this by forcing the following parameters when exporting
`-certpbe PBE-SHA1-3DES -keypbe PBE-SHA1-3DES -nomac`

Miracle then everything worked pretty well, until the next problem.
The impossibility of subsequently extracting the CRT or the KEY because OpenSSL refuses with this error message when I try to extract it
```
openssl pkcs12 -in CERTIFICATE.pfx -nokeys -out CERTIFICATE.crt -nodes
openssl pkcs12 -in CERTIFICATE.pfx -nokeys -out CERTIFICATE.cer -clcerts
Warning: MAC is missing!
```

I have tried an infinite number of commands to try to get around this problem but without success. My searches on Goo... were also not fruitful and as for my friend ChatGPT almost totally useless.

So I'm heading to the community to ask you to help me solve this problem because I really need these 2 functions

- Backwards compatibility
- The extraction capacity

Thank you in advance for your assistance

**Comment by t8m** [comment, marked as answer]
Why do you use `-nomac`? That should not be necessary for backwards compatibility with older Windows. You can try `-macalg sha1` instead.

**Reply by Prody33** [reply]
Thanks, I'll give it a try.

**Reply by Prody33** [reply]
Hello, I haven't had the time yet but I'm going to do it as soon as possible, I'll keep you informed.
Thanks again for your help

**Reply by Prody33** [reply]
Hello,
sorry for the late return, but I was on leave.
I can confirm that it works perfectly, so it's the hash that's the problem.
Thank you very much for your help
